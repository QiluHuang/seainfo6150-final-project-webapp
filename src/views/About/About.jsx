import React from 'react';
import styles from './About.module.css';
import DishImageLarge from '../../components/Images/DishImageLarge/DishImageLarge.jsx';

const About = () => {
  const imgUrl = [
    "https://d.newsweek.com/en/full/1552244/variety-chinese-food.jpg",
    "https://nitrocdn.com/QJsLnWfsWAiuukSIMowyVEHtotvSQZoR/assets/static/source/rev-d5d1d02/sites/default/files/styles/body_images_sm_1x/public/media/healthy-kung-pao-chicken-1.jpg",
    "https://purewows3.imgix.net/images/articles/2020_02/healthy-chinese-food-beef-and-broccoli.jpg?auto=format,compress&cs=strip"
  ];

    return (
        <div className={styles.container}>
          <div clasName={styles.info}>
            <div className={styles.wrapper}>
              <div className={styles.title}>Address</div>
              <div className={styles.text}>11760 San Pablo Ave CEl Cerrito, CA 94530</div>
            </div>
            <div className={styles.wrapper}>
              <div className={styles.title}>Phone</div>
              <div className={styles.text}>(510) 236-9999</div>
            </div>
            <div className={styles.wrapper}>
              <div className={styles.title}>Store Hours</div>
              <div className={styles.text}>
                <ul>
                  <li>Mon:	11:30 AM - 09:00 PM</li>
                  <li>Tue:	Closed</li>
                  <li>Wed - Thurs:	11:30 AM - 09:00 PM</li>
                  <li>Fri - Sat:	11:30 AM - 09:30 PM</li>
                  <li>Sun:	11:30 AM - 09:00 PM</li>
                </ul>
              </div>
            </div>
          </div>

          <div clasName={styles.image_wrapper}>
            <div><DishImageLarge url={imgUrl[0]} title={"chineseFood1"}/></div>
            <div><DishImageLarge url={imgUrl[1]} title={"chineseFood2"}/></div>
            {/* <div><DishImageLarge url={imgUrl[2]} title={"chineseFood3"}/></div> */}
          </div>
        </div>


        );
}

export default About;