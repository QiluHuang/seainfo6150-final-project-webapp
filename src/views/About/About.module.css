.container {
    margin-top: 5%;
    display: grid;
    grid-template-columns: 3fr 3fr;
}


.wrapper {
    display: flex;
    flex-direction: column;
    font-size: 1.3em;
    margin: 5% 5% 0% 20%;
}

.title {
    color: green;
    font-family: "Sofia", sans-serif;
    font-size: 1.2em;
    margin-bottom: 1%;
}





.text {
    font-family: Arial, Helvetica, sans-serif;
}

.image_wrapper {
    overflow: auto;
}



@media (max-width: 768px) {
    .container {
        margin-top: 5%;
        display: flex;
        flex-direction: column;
        transition-duration: 1s;

    }

    .wrapper {       
        font-size: 0.8em;
    }

    .img {
        margin-left: 30%;
    }
}
  
  @media (min-width: 768px) and (max-width: 1390px) {
    .wrapper {       
        font-size: 1em;
    }

    .image_wrapper {
        transition-duration: 1s;
        margin-top: 10%;
    }
}