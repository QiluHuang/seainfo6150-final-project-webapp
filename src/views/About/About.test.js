import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import About from "./About.jsx";

describe("About tests", () => {
    it("renders the About component correctly", () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();
    });
});