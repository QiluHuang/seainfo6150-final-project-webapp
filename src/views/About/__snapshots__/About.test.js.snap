// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About tests renders the About component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      clasname="info"
    >
      <div
        class="wrapper"
      >
        <div
          class="title"
        >
          Address
        </div>
        <div
          class="text"
        >
          11760 San Pablo Ave CEl Cerrito, CA 94530
        </div>
      </div>
      <div
        class="wrapper"
      >
        <div
          class="title"
        >
          Phone
        </div>
        <div
          class="text"
        >
          (510) 236-9999
        </div>
      </div>
      <div
        class="wrapper"
      >
        <div
          class="title"
        >
          Store Hours
        </div>
        <div
          class="text"
        >
          <ul>
            <li>
              Mon: 11:30 AM - 09:00 PM
            </li>
            <li>
              Tue: Closed
            </li>
            <li>
              Wed - Thurs: 11:30 AM - 09:00 PM
            </li>
            <li>
              Fri - Sat: 11:30 AM - 09:30 PM
            </li>
            <li>
              Sun: 11:30 AM - 09:00 PM
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      clasname="image_wrapper"
    >
      <div
        class="img"
      >
        <img
          alt="chineseFood1"
          class="img"
          src="https://d.newsweek.com/en/full/1552244/variety-chinese-food.jpg"
        />
      </div>
      <div
        class="img"
      >
        <img
          alt="chineseFood2"
          class="img"
          src="https://nitrocdn.com/QJsLnWfsWAiuukSIMowyVEHtotvSQZoR/assets/static/source/rev-d5d1d02/sites/default/files/styles/body_images_sm_1x/public/media/healthy-kung-pao-chicken-1.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
